project/src/main/java/com/google/sps/Servlets/Database.java

```diff
@@ -0,0 +1,85 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.text.SimpleDateFormat;  
+import java.util.Date;
+import java.util.ArrayList;
+import com.google.gson.*;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.PreparedQuery.TooManyResultsException;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.CompositeFilter;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
+import com.google.appengine.api.datastore.Query.Filter;
+import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.FilterPredicate;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+
+public class Database {
+
+  private static DatastoreService getDatastore() {
+    return DatastoreServiceFactory.getDatastoreService();
+  }
+
+  // public static List<int> getUserDocumentIDs(int id) {
+  //   getDatabase();
+  //   //return list of document ids
+  // }
+
+  private static User createUser(String email) {
+    Entity userEntity = new Entity("User");
+    ArrayList<String> documents = new ArrayList<String>();
+
+    userEntity.setProperty("email", email);
+    userEntity.setProperty("documents", documents);
+    getDatastore().put(userEntity);
+    long userID = userEntity.getKey().getId();
+
+    return new User(email, userID);
+  }
+
+  public static User getUserByID(long userID) {
+    Query query = new Query("User").addFilter("__key__", Query.FilterOperator.EQUAL, KeyFactory.createKey("User", userID));
+    Entity userEntity = getDatastore().prepare(query).asSingleEntity();
+
+    String email = (String) userEntity.getProperty("email");
+
```

Comment:
maybe add null check like you did in getUserByEmail (eg if cant find, return a new user)

Reply:
Good idea, I'll have to look into how to respond to a null user because the problem with creating a new user is that we won't actually have any of their info if the user doesn't exist. When a user doesn't exist with a certain email we can easily just create a new one because we just generate a new random uuid where as here we can't just pull an email out of thin air